doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,74 +1,170 @@
+such as with the :cmd:`Theorem` command.  It exits proof mode when
+you complete a proof, such as with the :cmd:`Qed` command.  Tactics,
+which are available only in proof mode, incrementally transform incomplete
+proofs to eventually generate a complete proof.
 
-Coq user interfaces usually have a way of marking whether the user has
-switched to proof editing mode. For instance, in coqtop the prompt ``Coq <``   is changed into
-:n:`@ident <`   where :token:`ident` is the declared name of the theorem currently edited.
+When you run Coq interactively, such as through CoqIDE, Proof General or
+coqtop, Coq shows the current proof state (the incomplete proof) as you
+enter tactics.  This information isn't shown when you run Coq in batch
+mode with `coqc`.
 
-At each stage of a proof development, one has a list of goals to
-prove. Initially, the list consists only in the theorem itself. After
-having applied some tactics, the list of goals contains the subgoals
-generated by the tactics.
+Proof State
+-----------
 
-To each subgoal is associated a number of hypotheses called the *local context*
-of the goal. Initially, the local context contains the local variables and
-hypotheses of the current section (see Section :ref:`gallina-assumptions`) and
-the local variables and hypotheses of the theorem statement. It is enriched by
-the use of certain tactics (see e.g. :tacn:`intro`).
+The :gdef:`proof state` consists of one
+or more unproven goals.
+Each goal has a :gdef:`conclusion` (the statement that is to be proven)
+and may have zero or more named facts that can be used in proving the
+conclusion.  The proof may also use facts from the :term:`local context`
+and the :term:`global context`, such as definitions and proven theorems.
 
-When a proof is completed, the message ``Proof completed`` is displayed.
-One can then register this proof as a defined constant in the
-environment. Because there exists a correspondence between proofs and
-terms of λ-calculus, known as the *Curry-Howard isomorphism*
-:cite:`How80,Bar81,Gir89,H89`, Coq stores proofs as terms of |Cic|. Those
-terms are called *proof terms*.
+The term ":gdef:`goal`" may refer to an entire goal or to the conclusion
+of a goal, depending on the context.
 
+The conclusion appears below a line and the facts appear above the line.
+The conclusion is a type and each fact has an associated type.
+Facts can be :term:`hypotheses <hypothesis>`
+(which are propositions), variables or local definitions.
+The name of the fact appears first.  The type is shown after the colon.  Local
+definitions are shown in the form `n := 0 : nat`, for example, in which `nat` is the
+type of `0`.
 
-.. exn:: No focused proof.
+The set of facts can be called the :gdef:`goal context`.  The :gdef:`local context`
+has local variables and hypotheses (see Section :ref:`gallina-assumptions`) defined in the current
+:ref:`section <section-mechanism>` and generally is taken to include the goal context(s).  The
```

Comment:
I think you can skip the "generally" here. Local context should always mean section declarations + goal context.

Reply:
Got it. Also tweaked the wording to make clear that the local context is with respect to a single goal.